graphql/resolvers.go

```diff
@@ -3262,8 +3273,50 @@ func (r *queryResolver) MainlineCommits(ctx context.Context, options MainlineCom
 		if mainlineCommitVersion.Version != nil || mainlineCommitVersion.RolledUpVersions != nil {
 			mainlineCommits.Versions = append(mainlineCommits.Versions, &mainlineCommitVersion)
 		}
-	}
 
+		// If we have reached the limit of versions we have checked lets fetch some more until we have satisfied the activated version limit
+		if index == len(versions)-1 {
+			var skipOrderNumber int
+
+			// If NextPageOrderNumber is nil then use the last version's RevisionOrderNumber
+			if mainlineCommits.NextPageOrderNumber == nil {
+				skipOrderNumber = versions[len(versions)-1].RevisionOrderNumber
+			} else {
+				skipOrderNumber = utility.FromIntPtr(mainlineCommits.NextPageOrderNumber)
+			}
+
+			opts = model.MainlineCommitVersionOptions{
+				Activated:       true,
+				Limit:           limit,
+				SkipOrderNumber: skipOrderNumber,
+			}
+			activatedVersions, err = model.GetMainlineCommitVersionsWithOptions(projectId, opts)
+			if err != nil {
+				return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("Error getting activated versions: %s", err.Error()))
+			}
+
+			opts.Activated = false
+			unactivatedVersions, err = model.GetMainlineCommitVersionsWithOptions(projectId, opts)
+			if err != nil {
+				return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("Error getting unactivated versions: %s", err.Error()))
+			}
+
+			newVersions := append(activatedVersions, unactivatedVersions...)
+			versions = append(versions, newVersions...)
+			sort.Slice(versions, func(i, j int) bool {
+				return versions[i].RevisionOrderNumber > versions[j].RevisionOrderNumber
+			})
+		}
+	}
+	// Gather some analytics data
+	endTime := time.Now()
+	grip.Info(message.Fields{
+		"message":                 "mainlineCommits",
+		"commitsChecked":          len(versions),
+		"timeTaken":               endTime.Sub(startTime).String(),
```

Comment:
Our resolvers offer an extension to show query duration in the http response. Do you think that could sufficient instead of logging additional details?

Reply:
We don't actually log that information anywhere, Its only used when testing locally. I wanted to log this additional information so i can understand how varying amounts of commits affect query performance and scalability.